test/e2e/main.test.ts

```diff
@@ -192,42 +205,50 @@ describe("E2E", function () {
     await cronKeeper.setUpkeepRefunder(topUp.address);
   });
 
+  describe("KeeperRegistry", () => {
+    it("should check upkeep", async () => {
+      const [upkeepNeeded, performData] = await keeperRegistry
+        .connect(ethers.constants.AddressZero)
+        .callStatic["checkUpkeep(uint256)"](upkeepId);
+
+      expect(upkeepNeeded).to.eq(true);
+      expect(performData).to.not.eq("0x");
+    });
+
+    it("should perform upkeep", async () => {
+      const [, performData] = await keeperRegistry
+        .connect(ethers.constants.AddressZero)
+        .callStatic["checkUpkeep(uint256)"](upkeepId);
+
+      const [success, gasUsed] = await keeperRegistry
+        .connect(ethers.constants.AddressZero)
+        .callStatic.simulatePerformUpkeep(upkeepId, performData);
+
+      expect(success).to.eq(true);
+      expect(gasUsed).to.be.gt(0);
+    });
+  });
+
   describe("Execute jobs", function () {
-    it("should have executed job successfully", async function () {
+    it("should execute job successfully", async function () {
       const [workableBefore] = await job.workable(networkName);
       expect(workableBefore).to.equal(true);
 
-      await chainlink.keeperRegistryPerformUpkeep(
-        registry,
-        registrySigners,
-        upkeepId,
-        chainlink.KeeperRegistryParams.F
-      );
+      const [, performData] = await cronKeeper.callStatic.checkUpkeep("0x00");
+      await cronKeeper.performUpkeep(performData);
 
       const [workableAfter, reasonBytes] = await job.workable(networkName);
       const reason = parseBytes32String(reasonBytes.padEnd(66, "0"));
 
       expect(reason).to.equal("Timer hasn't elapsed");
       expect(workableAfter).to.equal(false);
     });
-
-    it("should reduce upkeep balance after job execution", async function () {
-      const { balance: preBalance } = await registry.getUpkeep(upkeepId);
-      await chainlink.keeperRegistryPerformUpkeep(
-        registry,
-        registrySigners,
-        upkeepId,
-        chainlink.KeeperRegistryParams.F
-      );
-      const { balance: postBalance } = await registry.getUpkeep(upkeepId);
-      expect(preBalance).to.be.gt(postBalance);
-    });
```

Comment:
This test is outside of scope for this project. It would also require a lot of extra code to test it in the forked test setup. Instead some more test cases were added which are more relevant.